templates/code/logsToEs.js

```diff
@@ -21,18 +21,20 @@ exports.handler = function(input, context) {
         // parse the input from JSON
         var awslogsData = JSON.parse(buffer.toString('utf8'));
 
-        // transform the input to Elasticsearch documents
-        var elasticsearchBulkData = transform(awslogsData);
+        // transform the input to Elasticsearch documents array
+        var elasticsearchBulkArray = transform(awslogsData);
 
         // skip control messages
-        if (!elasticsearchBulkData) {
+        if (!elasticsearchBulkArray) {
             console.log('Received a control message');
             context.succeed('Control message handled successfully');
             return;
         }
 
+        var elasticsearchBulkPayload = esDocumentArrayToPayload(elasticsearchBulkArray);
+
         // post documents to the Amazon Elasticsearch Service
-        post(elasticsearchBulkData, function(error, success, statusCode, failedItems) {
+        post(elasticsearchBulkPayload, function(error, success, statusCode, failedItems) {
```

Comment:
This is changing what we are actually sending to elasticsearch, I don't think we want to do that.

Reply:
are you sure? I just returned a different format to this function and do the json with newlines transformations here instead of in that function so I can more easily iterate and get the failedItems

Reply:
Can we test it first?